storage/innobase/handler/ha_innodb.cc

```diff
@@ -7030,8 +7041,8 @@ ha_innobase::open(
 		if (UNIV_UNLIKELY(m_share->ib_table &&
 				  m_share->ib_table->is_corrupt &&
 				  srv_pass_corrupt_table <= 1)) {
-			free_share(m_share);
-
+			free_share_and_nullify(&m_share);
+			dict_table_close(ib_table, FALSE, FALSE);
```

Comment:
I think this should be
```suggestion
			dict_table_close(m_share->ib_table, FALSE, FALSE);
```

Reply:
Consider the intrinsic table opening, I would say ib_table is fine